File: codeforces/69.do_not_be_distracted!/dnbd.py
# Author: José Rodolfo (jric2002)
t = int(input())
while (t):
  n = int(input())
  tasks = str(input())
  temp = ""
  t_before = "."
  s = False
  for i in range(0, n):
    if (t_before != tasks[i]):
      for j in range(0, len(temp)):
        if (tasks[i] == temp[j]):
          s = True
          break
      if (not(s)):
        temp += tasks[i]
        t_before = tasks[i]
      else:
        break
  if (s):
    print("NO")
  else:
    print("YES")
  t -= 1

File: codeforces/69.do_not_be_distracted!/dnbd.cpp
#include <iostream>
#include <cstring>
/* Author: José Rodolfo (jric2002) */
using namespace std;
int main() {
  unsigned short int t, n;
  string tasks, temp;
  char t_before;
  bool s;
  cin >> t;
  while (t--) {
    cin >> n;
    cin >> tasks;
    temp = "";
    t_before = '.';
    s = false;
    for (int i = 0; i < n; i++) {
      if (t_before != tasks[i]) {
        for (int j = 0; j < temp.size(); j++) {
          if (tasks[i] == temp.at(j)) {
            s = true;
            break;
          }
        }
        if (!s) {
          temp += tasks[i];
          t_before = tasks[i];
        }
        else {
          break;
        }
      }
    }
    if (s) {
      cout << "NO" << endl;
    }
    else {
      cout << "YES" << endl;
    }
  }
  return 0;
}